tests/test_component.py

```diff
@@ -43,12 +46,52 @@ def _setup_component(
     )
 
 
+@pytest.mark.parametrize(
+    "repo_url",
+    [
+        "ssh://user@host/path/to/repo.git",
+        "user@host:path/to/repo.git",
+        "file:///path/to/repo.git",
+    ],
+)
+def test_component_setup_remote_no_sub(tmp_path, repo_url):
+    c = _setup_component(tmp_path, repo_url=repo_url)
+
+    pull_remote = c.repo.git.remote("get-url", "origin")
+    push_remote = c.repo.git.remote("get-url", "--push", "origin")
+    assert pull_remote == repo_url
+    assert push_remote == repo_url
+
+
+@pytest.mark.parametrize(
+    "repo_url,push_url",
+    [
+        ("http://host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://user@host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://user:pass@host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
```

Comment:
What happens to `http://example.com:1234/path/to/repo.git` or http://user:pass@example.com:1234/path/to/repo.git?

Reply:
I've refactored the implementation to drop everything but the host and path of the input URL, if the url starts with `http://` or `https://`.

Obviously we can't guess a potential non-standard SSH port for components which are served on a non-standard HTTP(S) port.